src/main/java/com/purbon/kafka/topology/actions/Action.java

```diff
@@ -9,6 +9,10 @@
 
   void run() throws IOException;
 
+  default void dryRun() throws IOException {
+    System.out.printf("No dryRun check specified for %s", this.getClass().getName());
```

Comment:
would the dry run function as default do 

```java
      outputStream.println(action);
```
wonder how we could handle this operation....

Reply:
Few options come to mind: String run() throws Exception - returning the log message.
More cumbersome, another non-default method **Action**, and supplying the default implementation in **BaseAction**.